File: csharp/03 - Lab 2 - Bugfix Dracula/VampireSlayer.Tests/HunterTest.cs
using System;
using Xunit;
using FluentAssertions;

namespace VampireSlayer.Tests
{
    public class HunterTest
    {
        [Fact]
        public void CanHunt_GivenDayTime_ThenCannotHunt()
        {
            Hunter hunter = new Hunter();
            hunter.CanHunt(new DateTime(2014, 4, 12, 16, 1, 0)).Should().BeFalse();
        }

        [Fact]
        public void CanHunt_GivenNightTime_ThenCanHunt()
        {
            Hunter hunter = new Hunter();

            hunter.CanHunt(new DateTime(2014, 4, 12, 1, 1, 0)).Should().BeTrue();
        }
        
    }
}

File: java/03 - Lab 2 - Bugfix Dracula/src/test/java/org/cegeka/vampireslayer/HunterTest.java
package org.cegeka.vampireslayer;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Calendar;
import java.util.Date;

import org.junit.Before;
import org.junit.Test;

public class HunterTest {
	private Hunter hunter;

	@Before
	public void setup() {
		hunter = new Hunter();
	}

	@Test
	public void canHunt_givenDayTime_thenCannotHunt() {
		Calendar calendar = Calendar.getInstance();
		calendar.set(2014, Calendar.APRIL, 12, 16, 1);
		Date huntingTime = calendar.getTime();

		assertThat(hunter.canHunt(huntingTime)).isFalse();
	}

	@Test
	public void canHunt_givenNightTime_thenCanHunt() {
		Calendar calendar = Calendar.getInstance();
		calendar.set(2014, Calendar.APRIL, 12, 1, 1);
		Date huntingTime = calendar.getTime();

		assertThat(hunter.canHunt(huntingTime)).isTrue();
	}

}
